workflow/executor/executor.go

```diff
@@ -461,6 +477,22 @@ func (we *WorkflowExecutor) stageArchiveFile(containerName string, art *wfv1.Art
 	}
 	// In the future, if we were to support other compression formats (e.g. bzip2) or options
 	// the logic would go here, and compression would be moved out of the executors
+	if strategy.Zip != nil {
+		fileName = fmt.Sprintf("%s.zip", art.Name)
+		localArtPath = filepath.Join(tempOutArtDir, fileName)
+		f, err := os.Create(localArtPath)
+		if err != nil {
+			return "", "", argoerrs.InternalWrapError(err)
+		}
+		zw := zip.NewWriter(f)
+		defer zw.Close()
+		err = archive.ZipToWriter(unarchivedArtPath, zw)
+		if err != nil {
+			return "", "", err
+		}
+		log.Infof("Successfully zipped %s to %s", unarchivedArtPath, localArtPath)
+		return fileName, localArtPath, nil
+	}
 	return fileName, localArtPath, nil
```

Comment:
Since this code seems to be a duplicate of above, I wonder if we should break it out into its own method or function?

Reply:
I'm a little confused if I pull out this part of the code and bring it to the front, the code style may appear inconsistent, please give me some suggestions for merging.

Reply:
> Since this code seems to be a duplicate of above, I wonder if we should break it out into its own method or function?

I'm a little confused if I pull out this part of the code and bring it to the front, the code style may appear inconsistent, please give me some suggestions for merging.

Reply:
Okay, let me look at it.

Reply:
I know what you mean, that the tgz code isn't broken into a separate function. I suppose it is only about 10 lines that are duplicated. If the approver's okay with it, it's okay with me. Thanks for your patience.

Reply:
Sorry for the delayed reply. I would prefer the separate function so if any bug is in this code, a developer can fix it in one place.
@juliev0 can you create techdept issue for other types?

Reply:
No problem. Once @Belyenochi makes his changes, I'll see what's left and create an issue for that. @Belyenochi if you have specific areas of concern regarding inconsistency let me know.

Reply:
@juliev0 I see, I'll try to decompose into a function.

Reply:
thanks!